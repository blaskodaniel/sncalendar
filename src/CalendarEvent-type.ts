import { ListItem } from '@sensenet/default-content-types'

export default class CalendarEvent extends ListItem {
  public Location?: string
  public StartDate?: string
  public EndDate?: string
  public Lead?: string
  public AllDay?: boolean
  public EventUrl?: string
  public OwnerEmail?: string
}
