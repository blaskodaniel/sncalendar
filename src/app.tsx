import React, { useContext } from 'react'
import { Container, CssBaseline, Grid } from '@material-ui/core'
import snLogo from './assets/sensenet_logo_transparent.png'
import { NavBarComponent } from './components/navbar'
import MainPanel from './components/mainpanel'
import { RepositoryContext } from './context/repository-provider'
import SharedProvider from './context/shared-context'

/**
 * The main entry point of your app. You can start h@cking from here ;)
 */
export const App: React.FunctionComponent = () => {
  const repo = useContext(RepositoryContext)
  repo.reloadSchema()

  return (
    <>
      <CssBaseline />
      <NavBarComponent />
      <Container
        maxWidth="lg"
        style={{
          width: '100%',
          minHeight: '80vh',
          display: 'flex',
          marginTop: '10px',
          verticalAlign: 'middle',
          alignItems: 'center',
          justifyContent: 'flex-start',
          flexDirection: 'column',
          backgroundPosition: 'center',
          backgroundRepeat: 'no-repeat',
          backgroundImage: `url(${snLogo})`,
          backgroundSize: 'auto',
        }}>
        <Grid container direction="column" justify="center">
          <Grid item xs={12} style={{ alignSelf: 'center' }}>
            <SharedProvider>
              <MainPanel />
            </SharedProvider>
          </Grid>
        </Grid>
      </Container>
    </>
  )
}
