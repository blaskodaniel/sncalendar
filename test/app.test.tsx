import { shallow } from 'enzyme'
import React from 'react'
import { App } from '../src/app'

describe('App', () => {
  it('App snapshot', () => {
    const l = shallow(<App />)
    expect(l).toMatchSnapshot()
  })
})
