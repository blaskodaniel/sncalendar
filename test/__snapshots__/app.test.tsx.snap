// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot 1`] = `
<Fragment>
  <CssBaseline />
  <Component />
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
    style={
      Object {
        "alignItems": "center",
        "backgroundImage": "url(sensenet_logo_transparent.png)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "auto",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginTop": "65px",
        "minHeight": "80vh",
        "verticalAlign": "middle",
        "width": "100%",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        style={
          Object {
            "alignSelf": "center",
          }
        }
        xs={12}
      >
        <SharedProvider>
          <MainPanel />
        </SharedProvider>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
